OrcScala/src/orc/Backend.scala

```diff
@@ -80,13 +90,26 @@ trait Compiler[+CompiledCode] {
     override def toURL = toURI.toURL
   }
 
+  class ManyCompilationExceptions(exceptions: Seq[Throwable]) extends CompilationException("Multiple compiler errors") {
+    exceptions.foreach(addSuppressed)
+  }
+
   /** Compile the code in the reader using the given options and produce error messages on the
     * err writer. This is a simple wrapper around the other compile function.
     */
-  @throws(classOf[IOException])
+  @throws(classOf[IOException]) @throws(classOf[CompilationException])
```

Comment:
I'm not sure I actually throw any exceptions of this kind. Obviously the typechecker throws, but I think those are caught. I'm not sure when or why I made this change. I'll revert it and sniff for throws.